### DISCUSSION TITLE
How can I set upstream when pushing to remote for the first time?

### DISCUSSION THREAD

**Discussion by Zamachi** [NeogitOrg/neogit#219] [General]
Hi! I'm not sure whether this is a bug or feature isn't implemented yet, but when I want to push a local branch to the remote, there is a `--set-upstream` flag, which I do enable (I hover over it, press <TAB> and it changes color, so I assume I've activated it) 
![image](https://user-images.githubusercontent.com/32929482/134897079-62516468-3666-4f65-8e15-95532248e32d.png)

And then when I try either `pushremote` or `upstream`, it doesn't work, `$` says error 128 and that's all.

![image](https://user-images.githubusercontent.com/32929482/134897116-6b977a66-2bb1-4e20-8a72-5e7c7e3dcb53.png)

Unless I *didn't* really activate the flag with <TAB>, there should be a prompt of sort when it detects I have no remote set?
I know of a way to manually set/remove a remote on Magit, but that option isn't available here *yet*, so am I doing something wrong or should I use Git bash to set the remote currently?

I'm on Windows 10 btw so that could also be the culprit...

**Comment by TimUntersberger** [comment]
Hmmm... I cant really reproduce this.

Also the `[stderr 2]` indicates that the you *should* see 2 lines when you toggle the command.

**Reply by Zamachi** [reply]
Have I done it the proper way, set the correct flags with <TAB>? 
I really cna't reproduce it now either cuz I've swapped machines, and I'm having new problems with my config that'll have to be sorted out before I give this one a shot again, I've set it manually in the git bash and it worked

Which 2 lines are you referring to btw?

**Reply by TimUntersberger** [reply]
> Have I done it the proper way, set the correct flags with ?

Yeah this should work.

> Which 2 lines are you referring to btw?

Pressing `tab` while over an history item should expand the item and show you the output, if there is any.
